module-apps/application-phonebook/models/MultipleNumbersModel.cpp

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include "MultipleNumbersModel.hpp"
+
+#include <application-phonebook/data/PhonebookStyle.hpp>
+#include <application-phonebook/data/PhonebookItemData.hpp>
+#include <application-phonebook/widgets/MultipleNumbersWidget.hpp>
+
+#include <service-appmgr/Controller.hpp>
+#include <ListView.hpp>
+
+MultipleNumbersModel::MultipleNumbersModel(app::ApplicationCommon *app) : application(app)
+{}
+
+auto MultipleNumbersModel::requestRecordsCount() -> unsigned int
+{
+    return internalData.size();
+}
+
+auto MultipleNumbersModel::getMinimalItemSpaceRequired() const -> unsigned int
+{
+    return phonebookStyle::inputLinesWithLabelWidget::h;
+}
+
+void MultipleNumbersModel::requestRecords(const uint32_t offset, const uint32_t limit)
+{
+    setupModel(offset, limit);
+    list->onProviderDataUpdate();
+}
+
+auto MultipleNumbersModel::getItem(gui::Order order) -> gui::ListItem *
+{
+    return getRecord(order);
+}
+
+void MultipleNumbersModel::createData(std::shared_ptr<ContactRecord> record)
+{
+    contactRecord = std::move(record);
+
+    auto firstNumber = new gui::MultipleNumbersWidget(utils::translate("app_phonebook_multiple_numbers_first"),
+                                                      contactRecord->numbers[0].number.getFormatted());
```

Comment:
consider using `.at()` instead of `[]` as it's more exception safe
(I know, there won't be any error with the first element in the array tbh :P )

Reply:
sure